icuSources/include/_foundation_unicode/uconfig.h

```diff
@@ -101,7 +101,7 @@
 #ifndef U_DISABLE_RENAMING
 #if APPLE_ICU_CHANGES
 //rdar://60884991 #58 Replace installsrc patching with changes directly in header files
-#define U_DISABLE_RENAMING 1
+#define U_DISABLE_RENAMING FOUNDATION_FRAMEWORK
```

Comment:
`FOUNDATION_FRAMEWORK ` implies Darwin but not the other way around because we can still run SwiftFoundation as a package on Darwin. In think here we actually want "not Darwin" since that's what `CMakeLists.txt` uses.

Reply:
Adjusted to `defined(__APPLE__)` which is the C/C++ equivalent check.